File: group-anagrams/group-anagrams.py
class Solution(object):
    def groupAnagrams(self, strs):
        """
        :type strs: List[str]
        :rtype: List[List[str]]
        """
        sorted_map = {}
        for val in strs:
            key = ''.join(sorted(val))
            if key not in sorted_map:
                sorted_map[key] = [val]
            else:
                sorted_map[key].append(val)
        
        return(sorted_map.values())

File: group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        map<string,vector<string>> mp;
        int n=strs.size();
        vector<vector<string>> v;
        for(int i=0;i<n;i++){
            string s=strs[i];
            sort(s.begin(),s.end());
            mp[s].push_back(strs[i]);
        }
        for(auto a:mp){
            std::cout << a.first;
            std::cout << ": ";
            for (auto b: a.second){
                std::cout << b << ", ";
            }
            std::cout << "\n";
        }
        for(auto a:mp){
            vector<string> temp=a.second;
            v.push_back(temp);
        }
        return v;
    }
};